src/ldc/eh/msvc.d

```diff
@@ -109,12 +111,16 @@ extern(C) void _d_throw_exception(Object e)
         if (!old_terminate_handler)
             old_terminate_handler = set_terminate(&msvc_eh_terminate);
     }
-    auto t = cast(Throwable) e;
-    exceptionStack.push(t);
+    
+    auto throwable = cast(Throwable) e;
+    exceptionStack.push(throwable);
+
+    if (throwable.info is null && cast(byte*)throwable !is typeid(throwable).init.ptr)
+        throwable.info = _d_traceContext();

```

Comment:
When constructing the exception, the user generally doesn't have any idea about where it's going to unwind to and whether there are catch handlers interested in the trace. `throw -> _d_throw_exception()` could check for registered handlers at runtime though, but it would also require compiler support to register the handlers when entering a `try`.